src/main/java/com/aws/greengrass/clientdevices/auth/iot/dto/CreateSessionDTO.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.iot.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+@Value
+@AllArgsConstructor
+public class CreateSessionDTO {
+    String thingName;
+    String certificatePem;
```

Comment:
Should we tie these credentials to a protocol (MQTT)? MQTT sessions are currently based on these fields: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/main/src/main/java/com/aws/greengrass/clientdevices/auth/session/MqttSessionFactory.java#L76-L88

Should we add them to this DTO?

Reply:
Well that is a detail the infra should hide a domain event doesn't need to know about mqtt